src/amberscript/parser.cc

```diff
@@ -515,34 +515,40 @@ Result Parser::ParseShaderBlock() {
     if (!token->IsIdentifier() && !token->IsString())
       return Result("expected virtual file path after VIRTUAL_FILE");
 
-    r = ValidateEndOfStatement("SHADER command");
-    if (!r.IsSuccess())
-      return r;
-
     auto path = token->AsString();
 
     std::string data;
     r = script_->GetVirtualFile(path, &data);
-    if (!r.IsSuccess()) {
+    if (!r.IsSuccess())
       return r;
-    }
 
     shader->SetData(data);
-  } else {
-    r = ValidateEndOfStatement("SHADER command");
+    shader->SetFilePath(path);
```

Comment:
nit: could you please add unit tests to check setting the file path?

Reply:
Done